api/lib/reducers/polkadotV0-reducers.js

```diff
@@ -428,30 +434,24 @@ function stakeDetailsReducer(network, message, reducers) {
   }
 }
 
-async function extractInvolvedAddresses(
+function extractInvolvedAddresses(
   lunieTransactionType,
   signer,
   message,
-  db
+  messageIndex,
+  blockEvents,
 ) {
   let involvedAddresses = []
   if (lunieTransactionType === lunieMessageTypes.SEND) {
     involvedAddresses = involvedAddresses.concat([signer, message.args[0]])
   } else if (lunieTransactionType === lunieMessageTypes.STAKE) {
     involvedAddresses = involvedAddresses.concat([signer], message.validators)
   } else if (lunieTransactionType === lunieMessageTypes.CLAIM_REWARDS) {
-    // the rewards claiming happens for all delegators so we get the rewards for the claimed era and extract all addresses from there
-    const dbRewards = (await db.getRewards(delegatorAddress)) || []
-    const involvedRewards = dbRewards.filter(({ height, validator }) => {
-      return height === String(message.args[1]) && validator === message.args[0]
-    })
-    involvedAddresses = involvedAddresses.concat(
-      [signer],
-      involvedRewards.map(({ address }) => address)
-    )
-    if (involvedRewards.length > 0) {
-      involvedAddresses = involvedAddresses.concat(involvedRewards[0].validator)
-    }
+    // we get all reward target addresses from block events, we don't include signer
```

Comment:
why not also include the sender?

Reply:
If the signer receives a reward it should be included in the events. Signer can be anyone who just trigger the reward distribution, doesn't need to be a reward target.